wai-extra/Network/Wai/Middleware/Gzip.hs

```diff
@@ -92,13 +94,23 @@ gzip set app env sendResponse = app env $ \res ->
         ResponseFile{} | gzipFiles set == GzipIgnore -> sendResponse res
         _ -> if "gzip" `elem` enc && not isMSIE6 && not (isEncoded res) && (bigEnough res)
                 then
-                    case (res, gzipFiles set) of
-                        (ResponseFile s hs file Nothing, GzipCacheFolder cache) ->
-                            case lookup hContentType hs of
-                                Just m
-                                    | gzipCheckMime set m -> compressFile s hs file cache sendResponse
-                                _ -> sendResponse res
-                        _ -> compressE set res sendResponse
+                    let runAction x = case x of
+                            (ResponseFile s hs file Nothing, GzipPreCompressed nextAction) ->
```

Comment:
I do not think the current (pre patch) code cares about `gzipFiles` setting for non `ResponseFile` cases.
The code checks only the request header, then if there is no `"gzip"` is shout-circuit to send response otherwise 
(as it already dissmised `GzipIgnore`) `compressE set res sendResponse` everything (but a `cacheFolder` case)